internal/executor/reporter/job_event_reporter_test.go

```diff
@@ -104,7 +106,51 @@ func TestJobEventReporter_SendsPreemptionEvent(t *testing.T) {
 	}
 }
 
-func setupTest(t *testing.T, existingPods []*v1.Pod) (EventReporter, *fakecontext.SyncFakeClusterContext, *job.JobRunStateStore, *FakeEventSender) {
+func TestJobEventReporter_SendsEventImmediately_OnceNumberOfWaitingEventsMatchesBatchSize(t *testing.T) {
+	jobEventReporter, eventSender, _ := setupBatchEventsTest(2)
+	pod1 := createPod(1)
+	pod2 := createPod(2)
+
+	jobEventReporter.QueueEvent(EventMessage{CreateSimpleJobFailedEvent(pod1, "failed", "cluster1", api.Cause_Error), util.ExtractJobRunId(pod1)}, func(err error) {})
+	time.Sleep(time.Millisecond * 100) // Allow time for async event processing
```

Comment:
I think the better way to do this would be to block until you've seen the expected event or some timeout has occurred.  That way you could set a long timeout (like 10 seconds) and in the common case (i.e. the test passes) you'll only be blocking for single digit millis.

That would probably take a bit of reworking because you'd need to do something like make the FakeEventSender put the events on a channel.  Happy to approve this as is and then we can rework it later.

Reply:
It is a bit awkward as you are waiting on something being added to an internal buffer - there isn't always going to be an external thing you can wait for

We could add a onEventProcessed() or something, you could then use in testing to wait on